From 364f49255e4dbb20e5dd32a05fdb4b662eab119f Mon Sep 17 00:00:00 2001
From: Richard Liu <r66033@freescale.com>
Date: Tue, 14 Jan 2014 17:58:10 +0800
Subject: [PATCH 2/2] ENGR00294018 fix apk install fail issue form web page

fix apk install fail issue form web page, root cause
is mimeType not correct so DownloadProvider not regard
it as apk and can't find matched activity to handle.

Signed-off-by: Richard Liu <r66033@freescale.com>
---
 .../com/android/webview/chromium/WebViewContentsClientAdapter.java    | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/chromium/java/com/android/webview/chromium/WebViewContentsClientAdapter.java b/chromium/java/com/android/webview/chromium/WebViewContentsClientAdapter.java
index 653c0b1..c1f5aa1 100644
--- a/chromium/java/com/android/webview/chromium/WebViewContentsClientAdapter.java
+++ b/chromium/java/com/android/webview/chromium/WebViewContentsClientAdapter.java
@@ -735,6 +735,10 @@ public class WebViewContentsClientAdapter extends AwContentsClient {
                     mimeType = "video/mp4";
             }
 
+            if(mimeType.equals("application/octet-stream") && url.endsWith(".apk")) {
+                mimeType = "application/vnd.android.package-archive";
+            }
+
             mDownloadListener.onDownloadStart(url,
                                               userAgent,
                                               contentDisposition,
-- 
1.8.0

